.github/workflows/nix.yaml

```
@@ -44,46 +52,61 @@
     outputs:
       docker-image-name: ${{ steps.docker.outputs.image_name }}
+    
     steps:
-      - uses: actions/checkout@v4
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      # Skip aarch64-linux builds if not on main branch
+      - name: Check build conditions
+        id: check-conditions
+        run: |
```

actionlint (1.7.4)

shellcheck reported issue in this script: SC2086:info:2:32: Double quote to prevent globbing and word splitting

(shellcheck)